sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+   *
+   * @param models the models to update
+   * @param manager the dynamic data base manager to use
+   */
+  DynModelAlgorithm(DynModelDefinitions& models, const inputs::DynamicDataBaseManager& manager);
+
+  /**
+   * @brief Perform the algorithm
+   *
+   * According to if the node is concerned in a macro connection, dispatch the node to update the macro connections in the models definition.
+   * Macro connections not connected to a network node will be discarded and not put in the dynamic models definitions
+   *
+   * @param node the node to process
+   */
+  void operator()(const NodePtr& node);
+
+ private:
+  /**
+   * @brief DynModel macro connect definition
+   */
+  struct MacroConnect {
+    /// @brief Default Constructor
+    MacroConnect() = default;
+
+    /**
+     * @brief Constructor
+     *
+     * @param idauto dynamic model id
+     * @param macroConnection macro connector id
+     */
+    MacroConnect(const DynModelDefinition::DynModelId& idauto, const DynModelDefinition::MacroConnection::MacroId& macroConnection) :
+        dynModelId(idauto),
+        macroConnectionId(macroConnection) {}
+
+    /**
+     * @brief Equality operator
+     * @param other the other macro connect to compare to
+     * @returns true if they are equal, false if not
+     */
+    bool operator==(const MacroConnect& other) const;
+
+    /**
+     * @brief Non-equality operator
+     * @param other the other macro connect to compare to
+     * @returns true if they are not equal, false if not
+     */
+    bool operator!=(const MacroConnect& other) const;
+
+    DynModelDefinition::DynModelId dynModelId;                       ///< automaton id
+    DynModelDefinition::MacroConnection::MacroId macroConnectionId;  ///< macro connection id
+  };
+
+  /**
+   * @brief Hash structure for macro connec
+   */
+  struct MacroConnectHash {
+    /**
+     * @brief Operator to hash a macro connect
+     * @returns unique hash
+     */
+    std::size_t operator()(const MacroConnect& connect) const noexcept;
+  };
+
+ private:
+  /**
+   * @brief Determines if a library is loadable in current environement
+   * @param lib library name
+   * @returns true if library could be loaded, false if not
+   */
+  static bool doesLibraryExist(const std::string& lib);
+
+  /**
+   * @brief Computes library path for library name
+   *
+   * Determines the library path with Dynawo API first then using DFL environement
+   *
+   * @param lib the library name
+   * @returns the filepath of the library, or nullopt if not found
+   */
+  static boost::optional<boost::filesystem::path> computeLibPath(const std::string& lib);
+
+ private:
+  /// @brief Extract models from configuration before processing the nodes
+  void extractDynModels();
+
+  /**
+   * @brief Process single association from configuratio
+   *
+   * @param automaton the config automaton definition
```

Comment:
the dynamic automaton config (same below)

Reply:
OK